src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -455,19 +450,6 @@ public static void ValidateDependencyGroups(SemanticVersion version, IEnumerable
             foreach (var dep in dependencies.SelectMany(s => s.Packages))
             {
                 PackageIdValidator.ValidatePackageId(dep.Id);
-
-                if (dep.VersionRange != null)
-                {
-                    if (dep.VersionRange.HasLowerBound && dep.VersionRange.MinVersion.IsSemVer2)
-                    {
-                        throw new InvalidOperationException(String.Format(CultureInfo.CurrentCulture, NuGetResources.SemVer2VersionsNotSupported, dep.VersionRange.MinVersion));
```

Comment:
This error message should mentioned that it's a dependency version, ideally also mentioning the dependency ID.